core/src/wallet/ripple/database/RippleLikeTransactionDatabaseHelper.cpp

```diff
@@ -48,9 +48,12 @@ namespace ledger {
             rowset<row> rows = (sql.prepare << "SELECT  tx.hash, tx.value, tx.time, "
                     " tx.sender, tx.receiver, tx.fees, tx.confirmations, "
                     "block.height, block.hash, block.time, block.currency_name "
+                    "memo.data, memo.fmt, memo.ty "
                     "FROM ripple_transactions AS tx "
                     "LEFT JOIN blocks AS block ON tx.block_uid = block.uid "
-                    "WHERE tx.hash = :hash", use(hash));
+                    "LEFT JOIN ripple_memos AS memo ON memo.transaction_uid = tx.transaction_uid "
```

Comment:
If you have one transaction and 10 fields in memo, this `select` will return you 10 rows, that will looks like 10 transactions. Right?

Reply:
Not really. Just look at the line 59. It doesn’t return 10 transactions. It returns 10 rows. Those rows are being consumed in `inflateTransaction`.

Reply:
I see.

Reply:
To explain further, from what I understand, it returns several rows that are _related_ to the same transaction — i.e. because of the `WHERE` — clause. And then the transaction is built by looping over the rows and setting fields there.

It’s always good to have eyes looking at that kind of code to be sure it’s not fucked up, so thank you. :)